### ISSUE TITLE
Converting the .csv to .xml 

### ISSUE THREAD

**Issue by m-kashani** [m-kashani/MS_Project#5]
- [ ] https://github.com/howl0893/custom-object-detection-datasets/blob/master/python/coco_json_utils.py

  - [ ] https://github.com/akTwelve/cocosynth/blob/master/docs/getting-started.md

 - [ ] https://github.com/howl0893/custom-object-detection-datasets

 - [ ] https://www.youtube.com/watch?v=jftZBfMZj8k&list=PLy-L-4xQRIuNm4tN9RIufCC6sqjuw1mAj&index=1

 - [ ] https://github.com/akTwelve/cocosynth/tree/cd9d583f6c4ce2c0fc185d8dedb4788b5b7af0f0

<hr></hr>
## To Do Next:

  - [ ] get it done after the While(evaluation, preprocessing ) + Comparison of Detectron2 output layer. -> I hardly couldn't. Very boring and not ... ( Not now.)

  - [ ] Alireza:
https://patrickwasp.com/create-your-own-coco-style-dataset/
So -> It's talking about the binary mask that I do not have yet, but it says that it's very easy to create.

* https://www.kaggle.com/julian3833/2-understanding-and-plotting-rle-bounding-boxes -> The new issue is creating the rle from BBOX coordination. (Later.)

![image](https://user-images.githubusercontent.com/3967516/76672337-449c6a00-6572-11ea-8dc4-c27bf52f3f55.png)

  - [ ] -> What about this? https://stackoverflow.com/questions/50368683/set-white-color-outside-boundingbox-python-opencv

  - [ ] https://discuss.pytorch.org/t/coco-dataset-from-custom-semantic-segmentation-dataset-for-detectron2/72266

# download, decompress the data
```bash
!wget https://github.com/matterport/Mask_RCNN/releases/download/v2.1/balloon_dataset.zip
!unzip balloon_dataset.zip > /dev/null
```

  - [ ] -> Checking out the balloon data-set again. + Annotation.
![image](https://user-images.githubusercontent.com/3967516/76692824-08c8d980-6632-11ea-98ee-ea1514c91999.png)
 
![image](https://user-images.githubusercontent.com/3967516/76692992-e7b5b800-6634-11ea-9c1c-c5bd8207d0df.png)

  - [ ] -> https://detectron2.readthedocs.io/modules/structures.html#detectron2.structures.BoxMode
  - [ ] -> going through the detectron issues349 :+1:
  - [x] Installing the : https://github.com/matterport/Mask_RCNN/issues/6
  - [x] https://github.com/facebookresearch/detectron2/issues/349 (Yess)
 

<hr></hr>

#### Sol 1. Email and make sure

  - [ ] ![image](https://user-images.githubusercontent.com/3967516/76260292-d54f0f00-622d-11ea-9722-cba375f4ab55.png)

<hr></hr>

#### Sol 2. Shortcut?

  - [ ] -install: https://github.com/microsoft/VoTT

#### Sol 3. PeerCode or out source

  - [ ] Peer Coding + Reverse engineering:
https://github.com/Shenggan/BCCD_Dataset/blob/master/BCCD/Annotations/BloodImage_00000.xml
  - [ ] https://github.com/Shenggan/BCCD_Dataset/blob/master/export.py `(reverse engineering)`

![image](https://user-images.githubusercontent.com/3967516/76245652-a1b2bb80-6212-11ea-8a59-deb763da6049.png)

![image](https://user-images.githubusercontent.com/3967516/76246725-a6786f00-6214-11ea-8072-280e27217f9b.png)

![image](https://user-images.githubusercontent.com/3967516/76246920-02db8e80-6215-11ea-8b47-f8aa87cfb587.png)

  - [ ] - https://detectron2.readthedocs.io/modules/structures.html#detectron2.structures.BoxMode

#  - [ ] - Code6.py is under construction.

<hr></hr>

Just generate a new code and get it done.

 - [ ] Prepare for coco dataset following this instruction [this instruction](https://github.com/facebookresearch/detectron2/tree/master/datasets).
https://github.com/facebookresearch/detectron2/tree/master/datasets

 - [ ] Avoid Training for now.
### [ImageNet](http://www.image-net.org/) Pretrained Models 

We provide backbone weights pretrained on ImageNet-1k dataset.
* [VoVNetV2-19-Slim-DW](https://www.dropbox.com/s/f3s7ospitqoals1/vovnet19_ese_slim_dw_detectron2.pth)
* [VoVNetV2-19-Slim](https://www.dropbox.com/s/8h5ybmi4ftbcom0/vovnet19_ese_slim_detectron2.pth)
* [VoVNetV2-19-DW](https://www.dropbox.com/s/9awvl0mxye3nqz1/vovnet19_ese_dw_detectron2.pth)
* [VoVNetV2-19](https://dl.dropbox.com/s/rptgw6stppbiw1u/vovnet19_ese_detectron2.pth)
* [VoVNetV2-39](https://dl.dropbox.com/s/q98pypf96rhtd8y/vovnet39_ese_detectron2.pth)
* [VoVNetV2-57](https://dl.dropbox.com/s/8xl0cb3jj51f45a/vovnet57_ese_detectron2.pth)
* [VoVNetV2-99](https://dl.dropbox.com/s/1mlv31coewx8trd/vovnet99_ese_detectron2.pth)

 - [ ] ?????? To train a model, run:
```bash
python /path/to/`model-detectron2`/train_net.py --config-file /path/to/`my_dataset`/configs/<config.yaml>
```

### [`COCO`](http://cocodataset.org/#home) Pretrained Models

https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md
* [1]
* [2]
* [3]
* [ ]
* [ ]

 - [ ] - https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/model_zoo.md

=== Object detection and recognition ===
https://en.wikipedia.org/w/index.php?title=List_of_datasets_for_machine-learning_research&action=edit&section=4

**Comment by m-kashani** [comment]
to do -> is crowd can be calculated by the histogram.
